pandas/io/parquet.py

```diff
@@ -115,16 +123,23 @@ def write(
                 **kwargs,
             )
         else:
-            self.api.parquet.write_table(
-                table, file_obj_or_path, compression=compression, **kwargs
-            )
+            self.api.parquet.write_table(table, path, compression=compression, **kwargs)
         if should_close:
             file_obj_or_path.close()
 
     def read(self, path, columns=None, **kwargs):
-        parquet_ds = self.api.parquet.ParquetDataset(
-            path, filesystem=get_fs_for_path(path), **kwargs
-        )
+        if is_fsspec_url(path) and "filesystem" not in kwargs:
+            import_optional_dependency("fsspec")
+            assert kwargs.get("use_legacy_dataset", True) is True
+            import fsspec.core
+
+            fs, path = fsspec.core.url_to_fs(path)
```

Comment:
does fsspec handle closing itself?

Reply:
The file system does not need to be closed, and files are closed when they fall out of scope

Reply:
ok